src/lsstseries/ensemble.py

```diff
@@ -284,6 +284,66 @@ def prune(self, threshold=50, col_name=None):
 
         return self
 
+    def bin_sources(self, time_window=1.0, additional_cols=None, use_map=True, **kwargs):
+        """Bin sources on within a given time range to improve the estimates.
+
+        Notes
+        -----
+        * This should only be used for slowly varying sources where we can
+        treat the source as constant within `time_window`.
+
+        * As a default the function only aggregates and keeps the id, band,
+        time, flux, and flux error columns. Additional columns can be preserved
+        by providing the mapping of column name to aggregation function with the
+        `additional_cols` parameter.
+
+        Parameters
+        ----------
+        time_window : `float` (optional)
+            The time range (in days) over which to consider observations in the same bin.
+        additional_cols : `dict` (optional)
+            A dictionary mapping column name to aggregation method.
+            Example: {"my_value_1": "mean", "my_value_2": "max"}
+        use_map : `boolean` (optional)
+            Determines whether `dask.dataframe.DataFrame.map_partitions` is
+            used (True). Using map_partitions is generally more efficient, but
+            requires the data from each lightcurve is housed in a single
+            partition. If False, a groupby will be performed instead.
+        """
+        # Bin the time and add it as a column.
+        tmp_time_col = "tmp_time_for_aggregation"
+        if tmp_time_col in self._source.columns:
+            raise KeyError(f"Column '{tmp_time_col}' already exists in source table.")
+        self._source[tmp_time_col] = self._source[self._time_col].apply(
+            lambda x: np.floor(x / time_window) * time_window, meta=pd.Series(dtype=float)
```

Comment:
I believe we want the purest mean of the times when the observations were taken. I am not sure that I understand the need for np.floor?

Reply:
The floor is only used for binning the time. The temporary time column will contain floor(time/window)*window.

If we use `times = [i/10.0 for i in range(20)]` and `time_window = 0.5`, we get:

```
times = [0.0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 1.7, 1.8, 1.9]
times / 0.5 = [0.0, 0.2, 0.4, 0.6, 0.8, 1.0, 1.2, 1.4, 1.6, 1.8, 2.0, 2.2, 2.4, 2.6, 2.8, 3.0, 3.2, 3.4, 3.6, 3.8]
floor(times / 0.5) = [0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 1.0, 1.0, 1.0, 1.0, 2.0, 2.0, 2.0, 2.0, 2.0, 3.0, 3.0, 3.0, 3.0, 3.0]
floor(times / 0.5) * 0.5 = [0.0, 0.0, 0.0, 0.0, 0.0, 0.5, 0.5, 0.5, 0.5, 0.5, 1.0, 1.0, 1.0, 1.0, 1.0, 1.5, 1.5, 1.5, 1.5, 1.5]
```
So floor(times / 0.5) * 0.5 becomes are bin labels.

The actual times are computed as the mean of all times in the bin.

Reply:
See my note above about this binning method though. I'm not sure this is what we want vs something more dynamic.